lib/services/access_checker.go

```diff
@@ -514,40 +514,41 @@ func (a *accessChecker) Traits() wrappers.Traits {
 
 // CheckDatabaseRoles returns whether a user should be auto-created in the
 // database and a list of database roles to assign.
-func (a *accessChecker) CheckDatabaseRoles(database types.Database) (create bool, roles []string, err error) {
-	// First, collect roles from this roleset that have "create_db_user" option.
+func (a *accessChecker) CheckDatabaseRoles(database types.Database) (mode types.CreateDatabaseUserMode, roles []string, err error) {
+	// First, collect roles from this roleset that have create database user mode set.
 	var autoCreateRoles RoleSet
 	for _, role := range a.RoleSet {
-		if role.GetCreateDatabaseUserOption() {
+		mode := role.GetCreateDatabaseUserMode()
+		if mode != types.CreateDatabaseUserMode_DB_USER_MODE_UNSPECIFIED && mode != types.CreateDatabaseUserMode_DB_USER_MODE_OFF {
```

Comment:
I've tried to keep the same behavior from using the `create_database_user` option. If there is a role (allow) with it set to true, it will always create the user.

We could update it to match this "pattern", but it would lead to breaking changes.